brat/Brat/Compile/Hugr.hs

```diff
@@ -387,8 +387,18 @@ compileWithInputs parent name = gets compiled <&> M.lookup name >>= \case
     -- This should only have one outgoing wire which leads to an `Id` node for
     -- the brat representation of the function, and that wire should have a
     -- function type
-    Prim _ -> error $ "TODO: Compiling a Prim node being used as a thunk, not directly Evaled."
-                      ++ " Should construct trivial 4-node DFG+I/O+CustomOp Hugr."
+    Prim (ext,op) -> do
+      let n = ext ++ ('_':op)
+      let [] = ins
+      let [(_, VFun Braty cty)] = outs
+      box_sig@(FunctionType inputTys outputTys) <- body <$> compileSig Braty cty
+      ((p, _ty), ()) <- compileConstDfg parent n box_sig $ \dfg_id -> do
+        ins <- addNodeWithInputs ("Inputs" ++ n) (OpIn (InputNode dfg_id inputTys)) [] inputTys
+        outs <- addNodeWithInputs n (OpCustom (CustomOp dfg_id ext op box_sig [])) ins outputTys
+        addNodeWithInputs ("Outputs" ++ n) (OpOut (OutputNode dfg_id outputTys)) outs [] >>= \case
+          [] -> pure ()
```

Comment:
I think the test that there's no outputs seems superfluous (+ haskell will emit a warning about the incomplete pattern match)

Reply:
Well, not superfluous exactly, but I don't think we need to be quite so defensive

Reply:
Yeah, this is pretty much the same thing as your other comment - it does seem dang hard to write what are basically assertions, so instead we just drop the value and lose a (small in terms of source code) check that things are working the way we expect. And the warnings against incomplete matches (there are loads in this file!) just require us to write much longer/more verbose code to turn one kind of runtime failure into another...

What's the right thing to do here? Can we `instance MonadFail Checking` and handle all these incomplete matches in our own code (which might still just `error` but at least it'd silence the warnings)? For another PR, obviously - I've shortened/silenced these here for the time being.